server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -104,11 +103,6 @@ private static boolean searchWithCollectorManager(
         return topDocsFactory.shouldRescore();
     }
 
-    @Override
-    public AggregationProcessor aggregationProcessor(SearchContext searchContext) {
-        return aggregationProcessor;
-    }
```

Comment:
I think we should revert this change and let `QueryPhaseSearcher` implementation provide the `AggregationProcessor` as well. Wrapper can just call the searcher instance to return the aggregationProcessor instead of having to decide which processor to use. Also in wrapper way it will be created per request which seems unnecessary